src/main/java/duke/CommandHandler.java

```diff
@@ -0,0 +1,220 @@
+package duke;
+
+import exceptions.EmptyCommandException;
+import exceptions.IllegalCommandException;
+import exceptions.MissingDescriptionException;
+import exceptions.UnknownCommandException;
+import jobs.Deadline;
+import jobs.Event;
+import jobs.Task;
+import jobs.ToDo;
+import messages.MessageFormat;
+import messages.MessageOptions;
+import messages.MessageWrapper;
+import java.util.ArrayList;
+
+public class CommandHandler {
+
+    private final MessageFormat format;
+    private final MessageWrapper wrapper;
+    private final CommandParser parser;
+    private String result = "";
+    private boolean isUpdatable = false;
+    private final ArrayList<Task> tasks = new ArrayList<>();
+
+    public CommandHandler(CommandParser parser, MessageFormat format, MessageWrapper wrapper){
+        this.format = format;
+        this.wrapper = wrapper;
+        this.parser = parser;
+    }
+
+    public boolean isPendingUpdate() {
+        return isUpdatable;
+    }
+
+    public void updateStatus(boolean success) {
+        isUpdatable = !success;
+    }
+
+    public String getCommandResult() {
+        return result;
+    }
+
+    public ArrayList<Task> getAddedTasks() {
+        return tasks;
+    }
+
+    private String[] getCommandStrings(ArrayList<Task> tasks){
+        String[] commands = new String[tasks.size()];
+        for(int i = 0; i < tasks.size(); i++){
+            commands[i] = tasks.get(i).toString();
+        }
+        return commands;
+    }
+
+    private static String[] getAddTaskMessage(String description, int count) {
+        return new String[]{
+                "Got it. I've added this task",
+                description,
+                formatListCount(count)
+        };
+    }
+
+    private static String[] getRemoveTaskMessage(String description, int count) {
+        return new String[]{
+                "Noted. I've removed this task:",
+                description,
+                formatListCount(count)
+        };
+    }
+
+    private static String formatListCount(int count) {
+        return "Now you have " + count + " tasks in the list.";
+    }
+
+    private static int getListedIndex(String parameter) {
+        if (parameter.length() != 1) {
+            return -1;
+        }
+        char character = parameter.toCharArray()[0];
+        if (Character.isDigit(character)) {
+            return Integer.parseInt(parameter) - 1;
+        }
+        if (Character.isLetter(character)) {
+            character = Character.toUpperCase(character);
+            return (int)character - Constants.LETTER_OFFSET - 1;
+        }
+        return -1;
+    }
```

Comment:
Consider using a switch statement instead

Reply:
It seems that Java does not support case Is statements (java case statements can only consist of values, unlike .NET languages that support evaluations of functions), so I think using a switch will make the code more repetitive and less readable. Would you mind teaching me how can I use switch statement without having these drawbacks? 
Thank you very much for your reply :)